internal/bminventory/inventory.go

```diff
@@ -258,7 +257,16 @@ type bareMetalInventory struct {
 	secretValidator validations.PullSecretValidator
 }
 
-var _ restapi.InstallerAPI = &bareMetalInventory{}
+func (b *bareMetalInventory) UpdateClusterInstallProgress(ctx context.Context, params installer.UpdateClusterInstallProgressParams) middleware.Responder {
+	c, err := b.getCluster(ctx, params.ClusterID.String())
+	if err != nil {
+		return common.GenerateErrorResponder(err)
+	}
+	if err := b.clusterApi.UpdateInstallProgress(ctx, c, params.ClusterProgress); err != nil {
```

Comment:
consider logging errors in the api level.

Reply:
done

snippet from how it looks now (good & bad case):
```golang
level=info msg="cluster 81e3eb15-a922-4b82-8e2a-cb1d91f12318 has been updated with the following updates [progress_progress_info Making more progress! progress_progress_updated_at 2020-12-09T13:49:11.415Z]" func=github.com/openshift/assisted-service/internal/cluster.updateClusterProgress file="/go/src/github.com/openshift/origin/internal/cluster/common.go:83" cluster_id=81e3eb15-a922-4b82-8e2a-cb1d91f12318 go-id=4746 pkg=cluster-state request_id=f0e29424-15e2-4261-902d-298fd0d5ffec


evel=error msg="Failed to update install progress" func="github.com/openshift/assisted-service/internal/bminventory.(*bareMetalInventory).UpdateClusterInstallProgress" file="/go/src/github.com/openshift/origin/internal/bminventory/inventory.go:266" error="no condition passed to run transition UpdateInstallationProgress from state pending-for-input" pkg=Inventory
```